src/backend/access/external/pxfuriparser.c

```diff
@@ -634,9 +638,20 @@ GPHDUri_parse_segwork(GPHDUri *uri, const char *uri_str)
 	segwork = strstr(uri_str, segwork_substring);
 	if (segwork == NULL)
 		return;
-
 	segwork += strlen(segwork_substring);
 
+	if (enable_secure_filesystem)
+	{
+		/* parse dfs address */
+		size_end = strchr(segwork, segwork_dfs_separator);
+		if(size_end != NULL)
```

Comment:
if it's a secured environment, if there is no dfs address data maybe we should error out. What do you think?